cpp/src/arrow/flight/sql/client.h

```diff
@@ -135,6 +135,19 @@ class ARROW_EXPORT FlightSqlClient {
   arrow::Result<std::unique_ptr<FlightInfo>> GetTableTypes(
       const FlightCallOptions& options);
 
+  /// \brief Request the information about all the data types supported.
+  /// \param[in] options          RPC-layer hints for this call.
+  /// \return The FlightInfo describing where to access the dataset.
+  arrow::Result<std::unique_ptr<FlightInfo>> GetTypeInfo(
+      const FlightCallOptions& options);
```

Comment:
@lidavidm Do you think we should make it more explicit that these public APIs are JDBC/ODBC specific, e.g. `GetXdbcTypeInfo`?

Reply:
Yes, let's do that and also rename the command `CommandGetXdbcTypeInfo` then. Especially as things don't quite correspond 1:1 with either API.

Reply:
Sounds better to have that name.

I've already made them on both implementation